core/src/main/java/org/apache/iceberg/RemoveSnapshots.java

```diff
@@ -359,10 +361,24 @@ private void removeExpiredFiles(List<Snapshot> snapshots, Set<Long> validIds, Se
                 }
               }
             });
-    deleteDataFiles(manifestsToScan, manifestsToRevert, validIds);
-    deleteMetadataFiles(manifestsToDelete, manifestListsToDelete);
+    if (cleanExpiredFiles) {
+      deleteDataFiles(manifestsToScan, manifestsToRevert, validIds);
+      deleteMetadataFiles(manifestsToDelete, manifestListsToDelete);
+    } else {
+      Set<String> filesToDelete = findFilesToDelete(manifestsToScan, manifestsToRevert, validIds);
+      expireSnapshotResult = ExpireSnapshotResult
+          .builder()
+          .addManifestFiles(manifestsToDelete)
+          .addManifestListFiles(manifestListsToDelete)
+          .addDataFiles(filesToDelete)
+          .build();
+    }
   }
 
+  public ExpireSnapshotResult getExpiredSnapshotResult() {
+    cleanExpiredSnapshots();
```

Comment:
Is there any way to avoid this side-effecting call in this function? Maybe Ryan's comment that there's a more appropriate noun is relevant as I don't typically expect a getter to have side effects.